src/tests/tribol_common_plane_penalty.cpp

```diff
@@ -594,6 +594,206 @@ TEST_F( CommonPlaneTest, common_plane_2d_interpen_check )
   compareGaps( couplingScheme, gap, 1.E-8, "kinematic_penetration" );
 }
 
+TEST_F( CommonPlaneTest, common_plane_viscous_tangential_2d )
+{
+  // this test has two edges with zero gap so there should be no contribution
+  // from the kinematic constraint enforcement, only the velocity based viscous
+  // tangential term
+  constexpr int numVerts = 2;
+
+  RealT x1[numVerts];
+  RealT y1[numVerts];
+  RealT x2[numVerts];
+  RealT y2[numVerts];
+
+  x1[0] = 1.0;
+  x1[1] = 0.0;
+  y1[0] = 0.0;
+  y1[1] = 0.0;
+
+  RealT gap = 0.01;
+  x2[0] = 0.0;
+  x2[1] = 1.0;
+  y2[0] = -gap;
+  y2[1] = -gap;
+
+  tribol::IndexT conn1[numVerts] = { 0, 1 };
+  tribol::IndexT conn2[numVerts] = { 0, 1 };
+
+  tribol::registerMesh( 0, 1, numVerts, &conn1[0], (int)( tribol::LINEAR_EDGE ), &x1[0], &y1[0], nullptr,
+                        tribol::MemorySpace::Host );
+  tribol::registerMesh( 1, 1, numVerts, &conn2[0], (int)( tribol::LINEAR_EDGE ), &x2[0], &y2[0], nullptr,
+                        tribol::MemorySpace::Host );
+
+  RealT fx1[numVerts] = { 0., 0. };
+  RealT fy1[numVerts] = { 0., 0. };
+  RealT fx2[numVerts] = { 0., 0. };
+  RealT fy2[numVerts] = { 0., 0. };
+
+  // set nodal velocities in <1,1> and <-1,-1> directions
+  RealT vx1[numVerts] = { 1., 1. };
+  RealT vy1[numVerts] = { 1., 1. };
+  RealT vx2[numVerts] = { -1., -1. };
+  RealT vy2[numVerts] = { -1., -1. };
+
+  tribol::registerNodalResponse( 0, &fx1[0], &fy1[0], nullptr );
+  tribol::registerNodalResponse( 1, &fx2[0], &fy2[0], nullptr );
+
+  tribol::registerNodalVelocities( 0, &vx1[0], &vy1[0], nullptr );
+  tribol::registerNodalVelocities( 1, &vx2[0], &vy2[0], nullptr );
+
+  tribol::setKinematicConstantPenalty( 0, 1. );
+  tribol::setKinematicConstantPenalty( 1, 1. );
+
+  RealT visc_coeff = 0.5;
+  tribol::registerRealElementField( 0, tribol::VISCOUS_DAMPING_COEFF, &visc_coeff );
+  tribol::registerRealElementField( 1, tribol::VISCOUS_DAMPING_COEFF, &visc_coeff );
+
+  tribol::registerCouplingScheme( 0, 0, 1, tribol::SURFACE_TO_SURFACE, tribol::NO_CASE, tribol::COMMON_PLANE,
+                                  tribol::VISCOUS_TANGENTIAL, tribol::PENALTY, tribol::BINNING_GRID,
+                                  tribol::ExecutionMode::Sequential );
+
+  tribol::setPenaltyOptions( 0, tribol::KINEMATIC, tribol::KINEMATIC_CONSTANT );
+  tribol::setContactAreaFrac( 0, 1.e-12 );
+
+  RealT dt = 1.;
+  int update_err = tribol::update( 1, 1., dt );
+
+  EXPECT_EQ( update_err, 0 );
+
+  tribol::CouplingSchemeManager& couplingSchemeManager = tribol::CouplingSchemeManager::getInstance();
+
+  tribol::CouplingScheme* couplingScheme = &couplingSchemeManager.at( 0 );
+
+  EXPECT_EQ( 1, couplingScheme->getNumActivePairs() );
+
+  // check forces
+  RealT force_y = 0.5 * gap / numVerts;
```

Comment:
Where does this come from?  Shouldn't the visc_coeff be a part of it?  Why is the gap a part of it?  Isn't the viscosity only dependent on the tangential relative velocity?